include/aws/common/json.h

```diff
@@ -0,0 +1,325 @@
+ */
+AWS_COMMON_API
+int aws_json_value_get_number(const struct aws_json_value *node, double *output);
+
+/**
+ * Gets the boolean of a boolean aws_json_value.
+ * @param node The boolean aws_json_value.
+ * @return The boolean of the boolean aws_json_value, otherwise false.
+ */
+AWS_COMMON_API
+int aws_json_value_get_boolean(const struct aws_json_value *node, bool *output);
+// ====================
+
+// ====================
+// Object API
+
+/**
+ * Adds a aws_json_value to a object aws_json_value.
+ * @param object The object aws_json_value you want to add a node to.
+ * @param key The key to add the aws_json_value at.
+ * @param node The aws_json_value you want to add.
+ * @return AWS_OP_SUCCESS if adding was successful, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_object_add(const struct aws_json_value *object, const struct aws_byte_cursor *cursor,
+                        const struct aws_allocator *allocator, const struct aws_json_value *node);
+
+/**
+ * Returns the aws_json_value at the given key.
+ * @param object The object aws_json_value you want to get the node from.
+ * @param key The key that the aws_json_value is at. Is case sensitive.
+ * @return The aws_json_value at the given key, otherwise NULL.
+ */
+AWS_COMMON_API
+struct aws_json_value *aws_json_object_get(const struct aws_json_value *object, const struct aws_byte_cursor *cursor,
+                                           const struct aws_allocator *allocator);
+
+/**
+ * Returns the aws_json_value at the given case insensitive key.
+ * @param object The object aws_json_value you want to get the node from.
+ * @param key The key that the aws_json_value is at. Is not case sensitive.
+ * @return The aws_json_value at the given key, otherwise NULL.
+ */
+AWS_COMMON_API
+struct aws_json_value *aws_json_object_get_insensitive(const struct aws_json_value *object,
+                                                       const struct aws_byte_cursor *cursor,
+                                                       const struct aws_allocator *allocator);
+
+/**
+ * Checks if there is a aws_json_value at the given key.
+ * @param object The object aws_json_value you want to check a key in.
+ * @param key The key that you want to check. Is case sensitive.
+ * @return AWS_OP_SUCCESS if a aws_json_value is found, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_object_has(const struct aws_json_value *object, const struct aws_byte_cursor *cursor,
+                        const struct aws_allocator *allocator);
+
+/**
+ * Removes the aws_json_value at the given key.
+ * @param object The object aws_json_value you want to remove a aws_json_value in.
+ * @param key The key that the aws_json_value is at. Is case sensitive.
+ * @return AWS_OP_SUCCESS if the aws_json_value was removed, otherwise AWS_OP_ERR.
+ */
+AWS_COMMON_API
+int aws_json_object_remove(const struct aws_json_value *object, const struct aws_byte_cursor *cursor,
```

Comment:
I'm not sure. The case I can see for having it is for data manipulation, like if you need to update a value in the object. Outside of that though, I do not think it would be needed.